torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -214,12 +214,32 @@ def parallelize_llama(model, world_mesh, parallel_dims, job_config: JobConfig):
             param_dtype=torch.bfloat16, reduce_dtype=torch.float32
         )
         ac_mode = job_config.activation_checkpoint.mode
+        # specifically compile just the RMSNorm layers
+        enable_compile_rmsnorm = job_config.training.compile_rmsnorm
+        if job_config.training.compile and enable_compile_rmsnorm:
+            logger.info(
+                "Entire model is compiled with torch.compile, disabling RMSNorm compilation"
+            )
+            enable_compile_rmsnorm = False
+        if enable_compile_rmsnorm:
+            logger.info("Compiling RMSNorm layers with torch.compile")
+
         fsdp_config = {"mesh": dp_mesh, "mp_policy": mp_policy}
+
         for layer_id, transformer_block in enumerate(model.layers):
             if job_config.activation_checkpoint.mode in ("full", "selective"):
                 transformer_block = checkpoint_wrapper(
                     transformer_block, job_config.activation_checkpoint
                 )
+
+            if enable_compile_rmsnorm:
+                transformer_block.attention_norm = torch.compile(
+                    transformer_block.attention_norm, dynamic=False
+                )
+                transformer_block.ffn_norm = torch.compile(
+                    transformer_block.ffn_norm, dynamic=False
+                )
```

Comment:
This is actually not correct. If AC is enabled, you will see 2 extra submodules in the AC module, `attention_norm`, `ffn_norm`. And these 2 submodules will never be called.